/*
 * This file was automatically generated by EvoSuite
 * Fri Apr 07 20:02:32 KST 2017
 */

package org.prevayler.demos.scalability.prevayler;

import static org.evosuite.runtime.EvoAssertions.assertThrownBy;
import org.junit.AfterClass;
import org.junit.BeforeClass;
import static org.junit.Assert.assertFalse;
import static org.junit.Assert.fail;

import java.math.BigDecimal;
import java.util.Date;

import org.evosuite.runtime.EvoRunner;
import org.evosuite.runtime.EvoRunnerParameters;
import org.evosuite.runtime.TooManyResourcesException;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.prevayler.demos.scalability.Record;
import org.prevayler.demos.scalability.RecordIterator;

@RunWith(EvoRunner.class) @EvoRunnerParameters(useVNET = true) 
public class TransactionSystem_ESTest2 extends TransactionSystem_ESTest_scaffolding2 {
@BeforeClass
public static void initEvoSuiteFramework() {
String a = "aa";
}
@AfterClass
public static void clearEvoSuiteFramework() {
String a = "aa";
}

  @Test
  public void test0()  throws Throwable  {
      TransactionSystem transactionSystem0 = new TransactionSystem();
      RecordIterator recordIterator0 = new RecordIterator(100000);
      // Undeclared exception!
      try { 
        transactionSystem0.replaceAllRecords(recordIterator0);
        fail("Expecting exception: TooManyResourcesException");
      
      } catch(TooManyResourcesException e) {
         //
         // Loop has been executed more times than the allowed 10000
         //
         assertThrownBy("org.evosuite.runtime.LoopCounter", e);
      }
  }

  @Test
  public void test1()  throws Throwable  {
      TransactionSystem transactionSystem0 = new TransactionSystem();
      // Undeclared exception!
      try { 
        transactionSystem0.replaceAllRecords((RecordIterator) null);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
      }
  }

  @Test
  public void test2()  throws Throwable  {
      TransactionSystem transactionSystem0 = new TransactionSystem();
      Record record0 = new Record(0L);
      // Undeclared exception!
      try { 
        transactionSystem0.performTransaction(record0, (Record) null, 0L);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
      }
  }

  @Test
  public void test3()  throws Throwable  {
      TransactionSystem transactionSystem0 = new TransactionSystem();
      Date date0 = new Date();
      Record record0 = new Record(0L, " $_No#PTctULoJ3/", " $_No#PTctULoJ3/", (BigDecimal) null, (BigDecimal) null, date0, date0);
      transactionSystem0.performTransaction(record0, record0, (-5099L));
      // Undeclared exception!
      try { 
        transactionSystem0.hashCode();
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         assertThrownBy("org.prevayler.demos.scalability.Record", e);
      }
  }

  @Test
  public void test4()  throws Throwable  {
      TransactionSystem transactionSystem0 = new TransactionSystem();
      RecordIterator recordIterator0 = new RecordIterator(1);
      transactionSystem0.replaceAllRecords(recordIterator0);
      assertFalse(recordIterator0.hasNext());
  }

  @Test
  public void test5()  throws Throwable  {
      TransactionSystem transactionSystem0 = new TransactionSystem();
      transactionSystem0.hashCode();
  }
}
